### PR TITLE
sharding: add integration testing for compression verifier

### PR DESCRIPTION
**Pull request by hkdsun**
Add some basic integration testing for https://github.com/Shopify/ghostferry/pull/57

@Shopify/pods

### CODE REVIEW

File: sharding/test/compression_verifier_test.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	"database/sql"
+	"testing"
+
+	"github.com/Shopify/ghostferry"
+	sth "github.com/Shopify/ghostferry/sharding/testhelpers"
+	"github.com/Shopify/ghostferry/testhelpers"
+	"github.com/stretchr/testify/suite"
+)
+
+const ()
```
Comment (fjordan) [inline sharding/test/compression_verifier_test.go]: can this be removed?

File: sharding/test/compression_verifier_test.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+	t.Require().NotEqual(testhelpers.TestCompressedData3, testhelpers.TestCompressedData4)
+
+	t.InsertCompressedRowInDb(43, "gftest1", testhelpers.TestCompressedData3, t.Ferry.Ferry.SourceDB)
+	t.InsertCompressedRowInDb(43, "gftest2", testhelpers.TestCompressedData4, t.Ferry.Ferry.TargetDB)
+
+	err := t.Ferry.Start()
+	t.Require().Nil(err)
+
+	t.Ferry.Run()
```
Comment (fjordan) [inline sharding/test/compression_verifier_test.go]: Should we be instantiating and checking an `ErrorHandler` in addition to checking the error returned from `t.Ferry.Start()` like the test above to ensure no errors are returned during verification?

Comment (hkdsun) [reply sharding/test/compression_verifier_test.go]: The default ErrorHandler panics if there's an error in the Ferry, not the greatest but it works and matches production

File: sharding/test/compression_verifier_test.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+	t.Require().Nil(err)
+
+	errHandler := &testhelpers.ErrorHandler{}
+	t.Ferry.Ferry.ErrorHandler = errHandler
+
+	t.Ferry.Run()
+
+	t.Require().NotNil(errHandler.LastError)
+	t.Require().Equal("verifier detected data discrepancy: verification failed on table: gftest1.test_compressed_table_1 for pks: 43", errHandler.LastError.Error())
+}
```
Comment (fw42) [inline sharding/test/compression_verifier_test.go]: nitpick: missing newline here

File: sharding/test/compression_verifier_test.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	t.InsertCompressedRowInDb(43, "gftest2", testhelpers.TestCompressedData4, t.Ferry.Ferry.TargetDB)
+
+	err := t.Ferry.Start()
+	t.Require().Nil(err)
+
+	t.Ferry.Run()
+}
+
+func (t *CompressionVerifierTestSuite) InsertCompressedRowInDb(id int, schema, data string, db *sql.DB) {
+	_, err := db.Exec("INSERT INTO "+schema+"."+testhelpers.TestCompressedTable1Name+" VALUES (?,?,2)", id, data)
```
[the remark this answers is not in the document]
Comment (hkdsun) [reply sharding/test/compression_verifier_test.go]: there's a tenant_id column, i'll make this more clear before merging 👍

**Pull request merged**